### PR TITLE
move specific option sub-struct from controller-manager into kube-controller manager packages

### PR DESCRIPTION
**Pull request by stewart-yu**
**What this PR does / why we need it**:
As follow-up [PR](https://github.com/kubernetes/kubernetes/pull/63283), in case of it too big to review.

Move specific option sub-struct from `controller-manager` into `kube-controller manager` packages, which only used by `kube-controller manager`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/assign @sttts @deads2k 
/cc @hzxuzhonghu 
/hold
just for the finally two commit, will rebase commit after [PR.](https://github.com/kubernetes/kubernetes/pull/63283) get merged.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/options/options.go [hunk 1]
```diff
@@ -49,32 +49,37 @@ import (
...
-	ReplicaSetController             *cmoptions.ReplicaSetControllerOptions
-	ReplicationController            *cmoptions.ReplicationControllerOptions
-	ResourceQuotaController          *cmoptions.ResourceQuotaControllerOptions
-	SAController                     *cmoptions.SAControllerOptions
-	ServiceController                *cmoptions.ServiceControllerOptions
+	CloudProvider     *cmoptions.CloudProviderOptions
+	Debugging         *cmoptions.DebuggingOptions
+	GenericComponent  *cmoptions.GenericComponentConfigOptions
+	KubeCloudShared   *cmoptions.KubeCloudSharedOptions
+	ServiceController *cmoptions.ServiceControllerOptions
```
Comment (sttts) [inline cmd/kube-controller-manager/app/options/options.go]: @luxas out of curiosity: is this disabled somehow when an external cloud ctrl manager is running?

**Comment by sttts** [comment]
@deads2k approved?

File: cmd/kube-controller-manager/app/options/options.go (hunk 1 above)
Comment (luxas) [reply cmd/kube-controller-manager/app/options/options.go]: cc @andrewsykim 
@sttts IIRC, yes it is
@sttts might be interesting reading: https://docs.google.com/document/d/1YqsPzeG10E59vea5StKFM6W_xZq034eTaf1n649JQvs/edit?ouid=100233907167775907499&usp=docs_home

**Comment by stewart-yu** [comment]
/retest
kindly ping @deads2k  to review

**Comment by sttts** [comment]
@mikedanese @lavalamp is this approved?

File: cmd/kube-controller-manager/app/options/options.go (hunk 1 above)
Comment (andrewsykim) [reply cmd/kube-controller-manager/app/options/options.go]: yes, service controller does nothing if cloud provider is not set or set to `external`

Comment (andrewsykim) [reply cmd/kube-controller-manager/app/options/options.go]: If you're curious https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/service/service_controller.go#L225-L227

Comment (andrewsykim) [reply cmd/kube-controller-manager/app/options/options.go]: Also: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L345-L350

**Comment by stewart-yu** [comment]
rebased and add a `blank line` in import file,  /cc @sttts @deads2k

**Pull request merged**